ResearchKit/ActiveTasks/ORKTimedWalkStepViewController.m

```diff
@@ -0,0 +1,127 @@
+
+
+#import "ORKTimedWalkStepViewController.h"
+#import "ORKTimedWalkContentView.h"
+#import "ORKTimedWalkStep.h"
+#import "ORKPedometerRecorder.h"
+#import "ORKActiveStepViewController_Internal.h"
+#import "ORKNavigationContainerView_Internal.h"
+#import "ORKStepViewController_Internal.h"
+#import "ORKActiveStepView.h"
+
+
+double const kDistanceTrackingThreshold = 100.0;
+
+
+@interface ORKTimedWalkStepViewController ()
+
+@property (nonatomic, strong) NSMutableArray *samples;
+@property (nonatomic, strong) ORKTimedWalkContentView *timedWalkContentView;
+@property (nonatomic, assign) NSTimeInterval trialDuration;
+
+@end
+
+@implementation ORKTimedWalkStepViewController
+
+- (instancetype)initWithStep:(ORKStep *)step {
+    self = [super initWithStep:step];
+    
+    if (self) {
+        self.suspendIfInactive = YES;
+    }
+    
+    return self;
+}
+
+- (ORKTimedWalkStep *)timedWalkStep {
+    return (ORKTimedWalkStep *)self.step;
+}
+
+- (void)initializeInternalButtonItems {
+    [super initializeInternalButtonItems];
+    
+    self.internalDoneButtonItem = nil;
+    self.continueButtonTitle = ORKLocalizedString(@"BUTTON_DONE", nil);
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.timedWalkContentView = [ORKTimedWalkContentView new];
+    self.timedWalkContentView.image = [self timedWalkStep].image;
+    self.activeStepView.activeCustomView = self.timedWalkContentView;
+    self.activeStepView.stepViewFillsAvailableSpace = YES;
+    self.activeStepView.continueSkipContainer.continueEnabled = YES;
+    
+    self.timerUpdateInterval = 0.1f;
+}
+
+- (void)finish {
+    [super finish];
+    
+    [self goForward];
+}
+
+- (void)countDownTimerFired:(ORKActiveStepTimer *)timer finished:(BOOL)finished {
+    self.trialDuration = timer.runtime;
+    [super countDownTimerFired:timer finished:finished];
+}
+
+- (ORKStepResult *)result {
+    ORKStepResult *sResult = [super result];
+    
+    NSMutableArray *results = [NSMutableArray arrayWithArray:sResult.results];
+    
+    ORKTimedWalkResult *timedWalkResult = [[ORKTimedWalkResult alloc] initWithIdentifier:(NSString *__nonnull)self.step.identifier];
+    timedWalkResult.distance = [self timedWalkStep].distance;
+    timedWalkResult.timeLimit = [self timedWalkStep].stepDuration;
+    timedWalkResult.duration = self.trialDuration;
+    
+    [results addObject:timedWalkResult];
+    
+    sResult.results = [results copy];
+    
+    return sResult;
+}
+
+#pragma mark - ORKPedometerRecorderDelegate
+
+- (void)pedometerRecorderDidUpdate:(ORKPedometerRecorder *)pedometerRecorder {
+    double timedWalkDistance = [self timedWalkStep].distance;
+    if (timedWalkDistance >= kDistanceTrackingThreshold) {
+        double distanceInMeters = timedWalkDistance - pedometerRecorder.totalDistance;
+        BOOL isValid = pedometerRecorder.totalDistance > 0;
+        [(ORKTimedWalkContentView *)self.activeStepView.activeCustomView setDistanceInMeters:distanceInMeters visible:isValid];
```

Comment:
@julientherier I understand you are trying to create the best experience for the user. 

But we still think removing the "remaining distance" label is the right thing to do. Because, in most of use cases of this test, the inaccurate distance progress label would not help the user. This test really should ask the user to use the physical landmarks as the start point and end point.

Also we probably should discourage user to check the phone's screen while walking for user's safety.